io/io/src/TBufferMerger.cxx

```diff
@@ -73,9 +74,12 @@ void TBufferMerger::Push(TBufferFile *buffer)
 {
    {
       std::lock_guard<std::mutex> lock(fQueueMutex);
+      fBuffered += buffer->BufferSize();
       fQueue.push(buffer);
    }
-   fDataAvailable.notify_one();
+
+   if (fBuffered > fAutoSave)
```

Comment:
fBuffered can be modified by another thread while this test is executed.  So as such fBuffered need to be an atomic (and/or the test itself (recorded in bool local variable) should be executed within the lock).